### ISSUE TITLE
Unclear error message on incorrect single-qubit and classical bit register subscripting

### ISSUE THREAD

**Issue by delapuente** [Qiskit/qiskit-terra#915]
### Informations
**Classical Register Subscripting Error Code** raises an error for the final two lines of code:
```
from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
quantum_reg = QuantumRegister(2)
classical_reg_0 = ClassicalRegister(1)
classical_reg_1 = ClassicalRegister(1)
quantum_circuit = QuantumCircuit(quantum_reg, classical_reg_0, classical_reg_1)
# Apply Hadamard gate on qubits
quantum_circuit.h(quantum_reg)
quantum_circuit.measure(quantum_reg[0], classical_reg_0) quantum_circuit.measure(quantum_reg[1], classical_reg_1)
```

```
qiskit._qiskiterror.QISKitError:
'expected classical register'
```

It is not necessarily apparent to a user that the variable they have provided is not a classical register, and that it needs to be subscripted. 

**Quantum Register Subscripting Error Code** raises the following error for the final line of code when addressing a single qubit quantum register:
```
from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
quantum_reg_0 = QuantumRegister(1)
quantum_reg_1 = QuantumRegister(1)
classical_reg_0 = ClassicalRegister(1)
classical_reg_1 = ClassicalRegister(1)
quantum_circuit = QuantumCircuit(quantum_reg_0, quantum_reg_1, classical_reg_0,
classical_reg_1)
# Apply Hadamard gate on qubit
quantum_circuit.h(quantum_reg_0)
quantum_circuit.h(quantum_reg_1)
quantum_circuit.measure(quantum_reg_0, classical_reg_0[0])
```

```
QuantumRegister(1, 'q0') is not a tuple. A qubit should be formated as a tuple.
```

Having experience with various quantum programming APIs, we fell into the trap of omitting the [0] subscript. This was not apparent from the error message. 

### Suggested solutions
Recommend clearer error messages for these specific common API usage mistakes.

**Comment by Exferro** [comment]
I would love to fix this as my first issue, could anyone tell me please how could I?

**Comment by Exferro** [comment]
@ajavadia if it's not impolite from my side, could you please help me dealing with this as the "good first issue"?

**Comment by lukasszz** [comment]
I've tried to solve this in https://github.com/Qiskit/qiskit-terra/pull/1330 .

@Exferro what kind of help did you need? How did you start with the problem?

**Comment by ajavadia** [comment]
@Exferro sorry I just saw your comment here. Looks like @lukasszz took care of this issue. If you still would like to contribute, please feel free to take up another issue and ask if you need any help.

**Comment by Exferro** [comment]
@lukasszz Thank you for your question, but I do apologize, I didn't mean that I need some help dealing with issue (it didn't seem to be that hard). Instead, I thought that I had to obtain some kind of "ticket", or discussion, or foreword to start dealing with it since I hadn't previous contributions. 

Now I see that it's not necessary and I will be more brave next times)

@ajavadia It's okay, thank you very much, I will try to do my best and cope with another issue indeed)

**Comment by ajavadia** [comment]
The error message was fixed in #1330.

And actually after #1643, the particular example in this thread will not throw an error (i.e. it is possible to measure a quantum register into a classical bit, as long as the quanutm register has length 1).
